warehouse/jobs/runner.go

```diff
@@ -0,0 +1,373 @@
+			var wg sync.WaitGroup
+			wg.Add(1)
+			go func() {
+				//Is this correct or should I be passing ch as a variable to the anonymous go routine instead?
+				responses := <-ch
+				asyncWhJob.log.Info("Response recieved from the pgnotifier track batch")
+				var successfulAsyncJobs []AsyncJobPayloadT
+				for _, resp := range responses {
+					var jobID pgnotifier.AsyncOutput
+					err = json.Unmarshal(resp.Output, &jobID)
+					if err != nil {
+						asyncWhJob.log.Errorf("Unable unmarshal output %s", err)
+						//Will skip to next response, leaving the table row in executing stage.
+						// We need to program retry strategy, timeout strategy for each row in the asyncJobtable
+						continue
+					}
+					ind := getSinglePayloadFromBatchPayloadByTableName(&asyncjobpayloads, jobID.TableName)
+					if ind == -1 {
+						//Continues with the next payload leaving this row in the executing stage
+						asyncWhJob.log.Errorf("tableName is not found while updating the response after completing the trackasyncJob %s", jobID.TableName)
+						continue
+					}
+					if resp.Status == pgnotifier.AbortedState {
+						pkgLogger.Errorf("[WH]: Error in running async task %v", resp.Error)
+						sampleError := fmt.Errorf(resp.Error)
+						asyncWhJob.updateSingleAsyncJob(&asyncjobpayloads[ind], WhJobAborted, sampleError.Error())
+						continue
+					}
+					if resp.Status == pgnotifier.FailedState {
+						pkgLogger.Errorf("[WH]: Error in running async task %v", resp.Error)
+						sampleError := fmt.Errorf(resp.Error)
+						asyncWhJob.updateSingleAsyncJob(&asyncjobpayloads[ind], WhJobFailed, sampleError.Error())
+						continue
+					}
+					successfulAsyncJobs = append(successfulAsyncJobs, asyncjobpayloads[ind])
+
+				}
+				asyncWhJob.updateMultipleAsyncJobs(&successfulAsyncJobs, WhJobSucceeded, "")
+				wg.Done()
+			}()
+			wg.Wait()
+		}
+	}
+}
+
+//Queries the jobsDB and gets active async job and returns it in a
+func (asyncWhJob *AsyncJobWhT) getPendingAsyncJobs(ctx context.Context) ([]AsyncJobPayloadT, error) {
+	var asyncjobpayloads = make([]AsyncJobPayloadT, 0)
+	//Filter to get most recent row for the sourceId/destinationID combo and remaining ones should relegated to aborted.
+	query := fmt.Sprintf(
+		`select 
+	id,
+	sourceid,
+	destinationid,
+	status,
+	jobrunid,
+	COALESCE(starttime, ''),
+	created_at,
+	updated_at,
+	destination_type,
+	taskrunid,
+	tablename,
+	jobtype,
+	async_job_type,
+	namespace from %s where status='%s' LIMIT %d`, warehouseutils.WarehouseAsyncJobTable, WhJobWaiting, MaxBatchSizeToProcess)
+	rows, err := asyncWhJob.dbHandle.Query(query)
+	if err != nil {
+		return asyncjobpayloads, err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var asyncjobpayload AsyncJobPayloadT
+		var status string
+		var updated_at, created_at string
+		err = rows.Scan(
+			&asyncjobpayload.Id,
+			&asyncjobpayload.SourceID,
+			&asyncjobpayload.DestinationID,
+			&status,
```

Comment:
Since we are not using status, updated_at, created_at. Let's not keep then.

Reply:
Removed them.